app/models/opentofu_worker.rb

```diff
@@ -74,10 +74,32 @@ def unit_environment_variables
 
   def configure_service_worker_deployment(definition)
     super
-
+    # overwriting container port to be same as opentofu-runner service port i.e. in this case 6000
     definition[:spec][:template][:spec][:containers].first[:ports] = [{:containerPort => container_port}]
 
+    # ovewriting home directory to terraform home dir
+    env_var_array = definition[:spec][:template][:spec][:containers][0][:env]
+    var_name = "HOME"
+    updated_value = "/home/node"
+    env_var_array.each do |env_var_key_val|
+      if env_var_key_val[:name] == var_name
+        env_var_key_val[:value] = updated_value
+        break
+      end
+    end
```

Comment:
I'm fine changing this here but I think a good follow-up would be to allow overriding of a home_dir method so that the default_environment used that rather than `Rails.root.join("tmp").to_s}`

Reply:
You can use detect to find a hash value so you don't have to write your own loop/break logic
```suggestion
    env_var_array = definition[:spec][:template][:spec][:containers][0][:env]
    env_var_array.detect { |env| env[:name] == "HOME" }[:value] = "/home/node"
```

Reply:
Thanks... I have updated the PR with your suggested code